src/contracts/GPv2Settlement.sol

```diff
@@ -179,4 +188,54 @@ contract GPv2Settlement {
         // take 2^256 transactions to achieve!
         nonces[pair] = nonce;
     }
+
+    /// @dev Decodes an order from calldata bytes returning order parameters.
+    /// @param encodedOrder The order as encoded calldata bytes.
+    function decodeOrder(bytes calldata encodedOrder)
+        internal
+        pure
+        returns (
+            uint112 sellAmount,
+            uint112 buyAmount,
+            uint32 validTo,
+            uint112 tip,
+            uint8 flags,
+            uint112 executedAmount,
+            uint8 v,
+            bytes32 r,
+            bytes32 s
+        )
+    {
+        // NOTE: This is currently unnecessarily gas inefficient. Specifically,
+        // there is a potentially extraneuous check to the encoded order length
```

Comment:
Typo in my typo fix, sorry about that. :grin: ([extraneous](https://www.collinsdictionary.com/dictionary/english/extraneous))
```suggestion
        // there is a potentially extraneous check to the encoded order length
```